src/clj/fluree/db/storage.cljc

```diff
@@ -48,7 +48,8 @@
 (defn split-address
   "Splits `address` into the fully qualified storage method and local path."
   [address]
-  (str/split address #":(?!.*:)" 2))
+  (let [components (str/split address #":")]
+    [(str/join ":" (butlast components)) (last components)]))
```

Comment:
was there a bug how it was previously? this looks like it's doing a bunch of extra work to split the entire string only to immediately join it again.

Reply:
Yes, I've detailed the problem in the commit message of [ab50edc](https://github.com/fluree/db/pull/915/commits/ab50edcb7715b10dcc5c62e5b733bc3075fbfc9d). I've update this implementation to be more efficient.